netdev/vendors/devices/base.py

```diff
@@ -1,55 +1,54 @@
 """
-Base Class for using in connection to network devices
-
-Connections Method are based upon AsyncSSH and should be running in asyncio loop
+Base Device
 """
 
 import asyncio
 import re
 
-import asyncssh
-
-from netdev.exceptions import TimeoutError, DisconnectError
 from netdev.logger import logger
+from netdev.version import __version__
+from netdev import utils
+from netdev.connections import SSHConnection, TelnetConnection
 
 
 class BaseDevice(object):
-    """
-    Base Abstract Class for working with network devices
-    """
 
     def __init__(
-        self,
-        host=u"",
-        username=u"",
-        password=u"",
-        port=22,
-        device_type=u"",
-        timeout=15,
-        loop=None,
-        known_hosts=None,
-        local_addr=None,
-        client_keys=None,
-        passphrase=None,
-        tunnel=None,
-        pattern=None,
-        agent_forwarding=False,
-        agent_path=(),
-        client_version=u"netdev",
-        family=0,
-        kex_algs=(),
-        encryption_algs=(),
-        mac_algs=(),
-        compression_algs=(),
-        signature_algs=(),
+            self,
+            host=u"",
+            username=u"",
+            password=u"",
+            port=22,
+            protocol='ssh',
+            device_type=u"",
+            timeout=15,
+            telnet_port=23,

```

Comment:
I have an idea about it: we can set the port to None and change the logic:
if the protocol is 'ssh' and the port is 'None'-> set the port to default '22'
if the protocol is 'telnet' and the port is 'None'-> set the port to default '23'
if the protocol 'ssh' or 'telnet' but the port isn't  'None' (user set it to the particular value) -> we stay it user-defined

In this logic we need only the 'port' and the 'protocol' args and don't need the 'telnet_port` arg